internal/blobtesting/map.go

```diff
@@ -104,7 +115,13 @@ func (s *mapStorage) PutBlob(ctx context.Context, id blob.ID, data blob.Bytes, o
 
 	data.WriteTo(&b)
 
+	if s.limit >= 0 && s.totalBytes+int64(b.Len()) > s.limit {
+		return errors.Errorf("exceeded limit, unable to add %v bytes, currently using %v/%v", b.Len(), s.totalBytes, s.limit)
+	}
+
+	s.totalBytes -= int64(len(s.data[id]))
 	s.data[id] = b.Bytes()
+	s.totalBytes += int64(b.Len())
```

Comment:
nit; not an issue right now, but for consistency:

```suggestion
	s.totalBytes += int64(len(s.data[id]))
```